app.js

```diff
@@ -1,8 +1,70 @@
-function hello(name) {
-  const greeting = `hello ${name}`;
-  return greeting;
+const { bootstrap } = require("kaholo-plugin-library");
+const { GOOGLE_API_CLIENT_NAMES } = require("./consts");
+const { injectGoogleApiClients } = require("./google-service");
+const {
+  prepareStartSpreadsheetPayload, prepareAddSheetPayload,
+  prepareInsertRowPayload, prepareModifyAccessRightsPayloads,
+} = require("./payload-functions");
+
+async function startSpreadsheet({ sheets, drive }, params) {
+  const payload = prepareStartSpreadsheetPayload(params);
+  const { data: result } = await sheets.spreadsheets.create(payload);
+  if (params.authorizedUsers) {
+    const permissionPayloads = prepareModifyAccessRightsPayloads({
+      spreadsheetUrl: result.spreadsheetUrl,
+      sharing: "restricted",
+      editors: params.authorizedUsers,
+    });
+    const requests = permissionPayloads.map(
+      (permissionPayload) => drive.permissions.create(permissionPayload),
+    );
+    await Promise.all(requests);
+    const prependMessageWith = params.authorizedUsers.length > 1 ? `Users ${params.authorizedUsers.join(", ")} were` : `User ${params.authorizedUsers[0]} was`;
+    // TODO: Change console.error to console.info
+    // when it is fixed for console.info to print
+    // messages in the Activity Log
+    console.error(`${prependMessageWith} granted writer permissions to the spreadsheet ${result.spreadsheetId}`);
+  }
+  return result;
+}
+
+async function addSheet({ sheets }, params) {
+  const { data: result } = await sheets.spreadsheets.batchUpdate(prepareAddSheetPayload(params));
+  return {
+    spreadsheetId: result.spreadsheetId,
+    properties: result.replies[0].addSheet.properties,
+  };
+}
+
+async function insertRow({ sheets }, params) {
+  let result;
+  const payload = prepareInsertRowPayload(params);
+  if (params.row) {
+    result = await sheets.spreadsheets.values.update(payload);
+  } else {
+    result = await sheets.spreadsheets.values.append(payload);
+  }
+  return result.data;
+}
+
+async function modifyAccessRights({ drive }, params) {
+  // TODO: Implement overwrite feature
```

Comment:
What about this one?

Reply:
From KP-703 in Jira:
**Parameter Clobber Existing Users (boolean, default false) - if true, throw “Clobber Existing is not yet implemented. If Viewers, Commenters, or Editors are specified, they will be added. If empty, no change is made.” OR - if this is really easy to do and you understand how it should work then go ahead and implement it.**

If you'd like I can spend some additional time on this and implement it.

Reply:
Let's do that.
Please change the parameter name thou, `Clobber` sounds very informal (afaik). Please ask Kenny for a more formal synonym.